src/tracepoint_format_parser.cpp

```diff
@@ -143,16 +167,42 @@ std::string TracepointFormatParser::parse_field(const std::string &line)
     return "";
 
   int size = std::stoi(line.substr(size_pos + 5, size_semi_pos - size_pos - 5));
-  std::string field = line.substr(field_pos + 6, field_semi_pos - field_pos - 6);
+  std::string field = line.substr(field_pos + 6,
+                                  field_semi_pos - field_pos - 6);
   auto field_type_end_pos = field.find_last_of("\t ");
   if (field_type_end_pos == std::string::npos)
     return "";
   std::string field_type = field.substr(0, field_type_end_pos);
-  std::string field_name = field.substr(field_type_end_pos+1);
+  std::string field_name = field.substr(field_type_end_pos + 1);
 
   if (field_type.find("__data_loc") != std::string::npos)
   {
-    field_type = "int";
+    std::regex rgx(".*field:__data_loc\\s(\\w+)(\\[\\])?.*");
+    std::cmatch match;
+    if (std::regex_search(line.c_str(), match, rgx))
+    {
+      // handle char[] types
+      if (match.size() == 3 && match[1] == "char" && match[2] == "[]")
+      {
+        field_type = "const char *";
+        std::regex offsetRgx(".*offset:(\\d+);.*");
+        std::cmatch offsetMatch;
+        if (std::regex_search(line.c_str(), offsetMatch, offsetRgx))
+        {
+          field_name = "data_loc_" + field_name;
+          return "  " + field_type + " " + field_name + "; //offset " +
+                 offsetMatch.str(1) + "\n";
+        }
+      }
+      else
+      {
+        field_type = match[1];
+      }
+    }
+    else
+    {
+      field_type = "int";
+    }
     field_name = "data_loc_" + field_name;
```

Comment:
Maybe we should change this prefix to something we're more certain will be unique now that we're stripping it out later. `_bpftrace_data_loc_`? Whatever we go for, it can be extracted out to a constant that's shared between here and the clang parser to make sure they stay in sync